### PR TITLE
Move unified search to OCS api

### PR DESCRIPTION
**Pull request by nickvergessen**
cc @tobiasKaminsky 

As per https://github.com/nextcloud/server/pull/20916#discussion_r444448710 the UI is okay, so now we make this OCS so the mobile apps can use it too.

### CODE REVIEW

File: core/src/services/UnifiedSearchService.js [hunk 1]
```diff
@@ -60,8 +60,9 @@ export async function getTypes() {
  * @returns {Promise}
  */
 export function search(type, query, cursor) {
-	return axios.get(generateUrl(`/search/providers/${type}/search?term=${query}`), {
+	return axios.get(generateOcsUrl('search/providers', 2) + type + '/search', {
```
Comment (skjnldsv) [inline core/src/services/UnifiedSearchService.js:63]:
```suggestion
	return axios.get(generateOcsUrl('search', 2) + `providers/${type}/search`, {
```

Comment (skjnldsv) [inline core/src/services/UnifiedSearchService.js:63]:
```suggestion
	return axios.get(generateOcsUrl('search', 2) + `providers/${type}/search`, {
```

**Review by skjnldsv — CHANGES_REQUESTED** [review]

**Review by ChristophWurst — APPROVED** [review]

File: core/src/services/UnifiedSearchService.js (hunk 1 above)
Comment (nickvergessen) [reply core/src/services/UnifiedSearchService.js:63]: doesn't really matter and so far I always kept everthing inside that does not use vars, but well

Comment (skjnldsv) [reply core/src/services/UnifiedSearchService.js]: Syntax readability :)

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**